File: src/main/java/org/tindalos/principle/domain/coredetector/Detector.java
package org.tindalos.principle.domain.coredetector;

import org.tindalos.principle.domain.expectations.DesignQualityExpectations;


public interface Detector {
    
	CheckResult analyze(CheckInput checkInput);

	boolean isWanted(DesignQualityExpectations designQualityExpectations);

}

File: src/main/java/org/tindalos/principle/domain/coredetector/Detector.scala
package org.tindalos.principle.domain.coredetector

import org.tindalos.principle.domain.expectations.Expectations

trait Detector {

  def analyze(checkInput: AnalysisInput): AnalysisResult
  def isWanted(designQualityExpectations: Expectations): Boolean
  
  override def toString() = this.getClass().getSimpleName()

}